hazelcast-client-new/src/main/java/com/hazelcast/client/cache/impl/AbstractClientInternalCacheProxy.java

```diff
@@ -409,11 +556,35 @@ protected ClientInvocationFuture invoke(ClientMessage req, Data keyData, int com
         } catch (Exception e) {
             throw ExceptionUtil.rethrow(e);
         }
-        return new ClientDelegatingFuture<Boolean>(future, clientContext.getSerializationService(),
-                putIfAbsentResponseDecoder);
+        ClientDelegatingFuture delegatingFuture =
+                new ClientDelegatingFuture<Boolean>(future, clientContext.getSerializationService(),
+                                                    putIfAbsentResponseDecoder);
+        if (async && statisticsEnabled) {
+            delegatingFuture.andThen(new ExecutionCallback<Object>() {
```

Comment:
@sancar at first version if this PR, I had implemented as you mentioned. But then I thought that it is not good idea using thread pools (executor) for sync operations for just statistics since its overhead to overall system may be more that making it as sync.